src/compiler/emitter.ts

```diff
@@ -3052,6 +3054,17 @@ module ts {
                     write(": ");
                     // Even though this is stored as identifier treat it as an expression
                     // Short-hand, { x }, is equivalent of normal form { x: x }
+
+                    if (languageVersion < ScriptTarget.ES6) {
+                    // short-hand property can be initialized with renamed let binding
+                    // in this case use proper name
+                        var generatedName = getGeneratedNameForIdentifier(node.name, /*isValueOfShorthandPropertyAssignment*/ true);

```

Comment:
Boolean flag also signals that for a given node instead of trying to get symbol for declaration we should go directly to name resolution if we don't want to get the symbol for the property instead of a symbol for a binding that initialized this property